tests/unit/library/test_json_stringformat.py

```diff
@@ -0,0 +1,941 @@
+"""Adapted from https://github.com/json-schema-org/JSON-Schema-Test-Suite/tree/9fc880bfb6d8ccd093bc82431f17d13681ffae8e/tests/draft2020-12/optional/format"""
+
+import pytest
+import json
+from .test_json import generate_and_check, check_match_failure
+
+
+class TestDate:
+    schema = '{"$schema":"https://json-schema.org/draft/2020-12/schema","format":"date"}'
+
+    @pytest.mark.parametrize(
+        "target_str",
+        [
+            "12",  # all string formats ignore integers
+            "13.7",  # all string formats ignore floats
+            "{}",  # all string formats ignore objects
+            "[]",  # all string formats ignore arrays
+            "false",  # all string formats ignore booleans
+            "null",  # all string formats ignore nulls
+            '"1963-06-19"',  # a valid date string
+            '"2020-01-31"',  # a valid date string with 31 days in January
+            '"2021-02-28"',  # a valid date string with 28 days in February (normal)
+            '"2020-02-29"',  # a valid date string with 29 days in February (leap)
+            '"2020-03-31"',  # a valid date string with 31 days in March
+            '"2020-04-30"',  # a valid date string with 30 days in April
+            '"2020-05-31"',  # a valid date string with 31 days in May
+            '"2020-06-30"',  # a valid date string with 30 days in June
+            '"2020-07-31"',  # a valid date string with 31 days in July
+            '"2020-08-31"',  # a valid date string with 31 days in August
+            '"2020-09-30"',  # a valid date string with 30 days in September
+            '"2020-10-31"',  # a valid date string with 31 days in October
+            '"2020-11-30"',  # a valid date string with 30 days in November
+            '"2020-12-31"',  # a valid date string with 31 days in December
+            '"2020-02-29"',  # 2020 is a leap year
+        ],
+    )
+    def test_good(self, target_str):
+        schema_obj = json.loads(self.schema)
+        target_obj = json.loads(target_str)
+        generate_and_check(target_obj, schema_obj)
+
+    @pytest.mark.parametrize(
+        "bad_str",
+        [
+            '"2020-01-32"',  # a invalid date string with 32 days in January
+            pytest.param('"2021-02-29"', marks=pytest.mark.xfail(reason="number of days not yet tied to month")),  # a invalid date string with 29 days in February (normal)
```

Comment:
If you want some extra challenge, remember that 2000 was a leap year, but that 1900 wasn't ;-)